### ISSUE TITLE
Compilation troubles of TinyEmberPlus on Linux

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Lawo/ember-plus#33]
```
/home/arndt/src/emberplus-1.6.2-A/tinyember/TinyEmberPlus/gadget/DirtyState.h:19
: error:  shadows template parm 'class FlagType'
     template<typename FlagType>

That is due to violating C++11 Syntax
The relevant language specification is §14.6.1/7 A:
"A template-parameter shall not be redeclared within its scope (including 
nested scopes). A template-parameter shall not have the same name as the 
template name."
While the Microsoft Compiler seems to do it anyway, GCC is NOT.
```

Original issue reported on code.google.com by `nullable...@gmail.com` on 5 Dec 2014 at 9:17

**Comment by blackhearted** [comment]
I already have both TinyEmberPlus and TinyEmberPlusRouter buildable and runnable under Linux(gcc version 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04.1) ).
If you want I can submit the pull-requests which fix the compilation.
The only problem - I have no Visual Studio.

**Comment by KimonHoffmann** [comment]
Fixed by merging PR #43 and cherry-picking 7fb46b3, from PR #42. Thanks @blackhearted for both additions.
